hr_time/public/js/easy_checkin_dialog.js

```diff
@@ -1,129 +1,355 @@
+   * @returns {Array<Object>} - The fields configuration for Frappe's UI dialog.
+   */
+  getDialogFields(employee_id) {
+    return [
+      {
+        label: "Action",
+        fieldname: "action",
+        fieldtype: "Select",
+        options: this.options,
+        default: this.default,
+        change: () => this.updateDialogBasedOnAction(),
+      },
+      {
+        fieldtype: "Section Break",
+        depends_on: 'eval: doc.action === "End of work"',
+      },
+      {
+        fieldname: "worklog_section",
+        fieldtype: "HTML",
+        options: this.getWorklogSectionHTML(),
+      },
+      {
+        fieldname: "worklog_box",
+        fieldtype: "Text",
+        placeholder: __("Describe your task here") + "...",
+      },
+      {
+        fieldtype: "Button",
+        label: "✔",
+        fieldname: "add_log",
+        click: () => this.addWorklog(employee_id),
+      },
+    ];
+  }
+
+  /**
+   * Generates the HTML content for the worklog section of the dialog.
+   * @returns {string} - The HTML content for the worklog section.
+   */
+  getWorklogSectionHTML() {
+    return `<div class="d-flex justify-between">
+                <label id="worklog_section_label">${__("Add Worklog")}</label>
+                <button class="btn btn-outline-info btn-xs edit-full-form-btn">${__(
+                  "Enter complete detail"
+                )}
+                    &nearr;
+                </button>
+            </div>
+            `;
+  }
+
+  /**
+   * Updates the dialog UI based on the selected action.
+   * If "End of work" is selected and no worklogs exist, disables the submit button.
+   */
+  updateDialogBasedOnAction() {
+    let action_value = this.dialogUI.get_value("action");
+    const isEndOfWork = action_value === "End of work";
+    const isIncompleteCheckout = isEndOfWork && !this.hasWorklogs;
+
+    this.dialogUI.get_primary_btn().prop("disabled", isIncompleteCheckout);
+    this.dialogUI.get_field("worklog_section").$wrapper.toggle(isEndOfWork);
+
+    if (this.hasWorklogs) {
+      this.dialogUI.$wrapper
+        .find("label#worklog_section_label")
+        .removeClass("not-filled");
+      this.dialogUI.$wrapper
+        .find("label#worklog_section_label")
+        .addClass("filled");
+      this.dialogUI.$wrapper
+        .find('button[data-fieldname="add_log"]')
+        .removeClass("not-filled");
+      this.dialogUI.$wrapper
+        .find('button[data-fieldname="add_log"]')
+        .addClass("filled");
+    } else {
+      this.dialogUI.$wrapper
+        .find("label#worklog_section_label")
+        .addClass("not-filled");
+      this.dialogUI.$wrapper
+        .find("label#worklog_section_label")
+        .removeClass("filled");
+      this.dialogUI.$wrapper
+        .find('button[data-fieldname="add_log"]')
+        .addClass("not-filled");
+      this.dialogUI.$wrapper
+        .find('button[data-fieldname="add_log"]')
+        .removeClass("filled");
```

Comment:
Code style: Please try to use the full line width, which would makes reading this section easier.